### ISSUE TITLE
Validation Accuracy

### ISSUE THREAD

**Issue by mhsung** [tflearn/tflearn#574]
Hi,

If the validation dataset is the same with training dataset, my understanding is that the validation accuracy must be the same with training accuracy. Is this correct?

I'm testing with MNIST example, but it always gives different accuracy.
https://github.com/tflearn/tflearn/blob/master/examples/images/convnet_mnist.py
I changed the last line as follows:
model.fit({'input': X}, {'target': Y}, n_epoch=20,
           validation_set=({'input': X}, {'target': Y}),
           snapshot_step=100, show_metric=True, run_id='convnet_mnist')

And, this is the output.
--
Training Step: 100  | total loss: 0.36543 | time: 5.375s
| Adam | epoch: 001 | loss: 0.36543 - acc: 0.8961 | val_loss: 0.19358 - val_acc: 0.9418 -- iter: 06400/55000
--

What is the reason that validation and training accuracy are different for the same dataset?

**Comment by Legor** [comment]
It may be, that the first value listed there is only the accuracy for your last training batch of this epoch. Otherwise i would agree that it should be the same.

**Comment by SaadSen** [comment]
I've a similar question. Every time I train on a same data, why do I get different accuracy values? I'm not giving any validation data. Is there a way to get same loss/accuracy value for same data while training? I tried initializing a graph with fixed seed value...did't help. Is that even normal to get different values every time?

Thanks,

**Comment by Legor** [comment]
In my opinion in theory you should get the same output for each run if **all** parameters are the same. This forbids any random values in your network at any stage and also the order of the input data as presented to the network should be the same.

**Comment by TrinhTUHH** [comment]
I use [AlexNet](https://github.com/tflearn/tflearn/blob/master/examples/images/alexnet.py) model for flower classification. When I set `validation_set` to **None** meaning no validation data, why do I still got 'acc' value in the output? Is it true that when `validation_set` is None, the validation set is the training set itself?
